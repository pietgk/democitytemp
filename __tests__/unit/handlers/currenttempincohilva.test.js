const lambda = require('../../../src/handlers/currenttempincovilha');
const { insertMockEvent } = require('../../../src/storage/storage-insert-mock-event');
const stage = 'test';
const httpMethod = 'GET';

jest.setTimeout(20000); // 20 second timeout
// TODO see if we want in memory mongodb as unit test context 
// and this as integration test.
// currently i like testing the same as prod.

describe('currenttempincovilha', () => {
    it('should fetch the currenttempincovilha from the database events collection', async () => {
        const _id = 'currenttempincovilha';
        const mockEvent = { _id, city: 'covilha', temperature: '15', unit: 'C' };
        const mockInsertResult = await insertMockEvent({ stage, ...mockEvent });
        const event = {
            httpMethod,
            stage
        };
        const responce = await lambda.getEventHandler(event);
 
        expect(mockInsertResult.insertedCount).toEqual(1);
        expect(mockInsertResult.insertedId).toEqual(_id);
        expect(responce.statusCode).toEqual(200);
        expect(responce.result).not.toEqual("null");
        expect(responce).toMatchSnapshot();
    });
});
