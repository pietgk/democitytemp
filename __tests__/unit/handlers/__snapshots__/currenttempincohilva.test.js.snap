// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`currenttempincovilha should fetch the currenttempincovilha from the database events collection 1`] = `
Object {
  "body": "{\\"_id\\":\\"currenttempincovilha\\",\\"city\\":\\"covilha\\",\\"temperature\\":\\"15\\",\\"unit\\":\\"C\\"}",
  "statusCode": 200,
}
`;
